tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +91,494 @@ mvn package
 
 ### Database Server Configuration
 
-All configuration for communication with the database is stored in the `/src/main/resources/application.properties` file.  This includes the connection string, username, and password.  The default username is assumed to be `Administrator` and the default password is assumed to be `password`.  If these are different in your environment you will need to change them before running the application.
+- The `CouchbaseConfig` class is a Spring configuration class responsible for setting up the connection to a Couchbase database in a Spring Boot application. It defines two beans:
+
+- `getCouchbaseCluster()`: This bean creates and configures a connection to the Couchbase cluster using the provided hostname, username, and password.
+
+- `getCouchbaseBucket(Cluster cluster)`: This bean creates a Couchbase bucket within the cluster if it doesn't already exist and returns the Bucket object associated with the specified bucket name.
+
+### Application Properties
+
+You need to configure the connection details to your Couchbase Server in the application.properties file located in the src/main/resources directory.
+
+In the connection string, replace `DB_CONN_STR` with the connection string of your Couchbase cluster. Replace `DB_USERNAME` and `DB_PASSWORD` with the username and password of a Couchbase user with access to the bucket.
+
+The connection string should be in the following format:
 
-### Dependency Injection via DBSetupRunner Class
+```properties
+spring.couchbase.bootstrap-hosts=couchbases://cb.fwu-oviwcpjq4v.cloud.couchbase.com
```

Comment:
What about non localhost server installations? How does the connection string work? Linking to Spring Boot Docs would be sufficient.

Reply:
added links